lib/src/script.dart

```diff
@@ -153,6 +153,21 @@ class Script {
   /// script exits.
   final _outputCloser = StreamCloser<List<int>>();
 
+  /// The script's signal handler to terminate the process.
+  late final Future<FutureOr<bool> Function([ProcessSignal])> _signalHandler;
```

Comment:
Why make the signal argument optional?

Reply:
fits Process.kill, but I'll make it non optional for the internal type